### PR TITLE
added new spatial joins

### PR DESCRIPTION
**Pull request by arya-hemanshu**
### Description
Added new spatial joins ```equals, disjoint, intersects, touches, crosses, contains, overlaps, relate```

### Checklist
None

### CODE REVIEW

**Comment by lorenaqendro** [comment]
@arya-hemanshu @thanosbnt can you have a look, please?
Removed [`relate`]( https://postgis.net/docs/ST_Relate.html) from the list as it was not like the other joins, returning true or false. It might be more useful to have it as a join for the fields.

File: src/main/java/uk/org/tombolo/core/utils/SubjectUtils.java [hunk 1]
```diff
@@ -135,11 +136,13 @@ private static Query queryFromSubjectSpecification(Session session, SubjectRecip
...
-			if (subjectRecipe.getGeoMatchRule().geoRelation == SubjectRecipe.SubjectGeoMatchRule.GeoRelation.within){
-				hqlQuery += " and within(shape, :geom) = true";
+			List<SubjectRecipe.SubjectGeoMatchRule.GeoRelation> geoRel = new ArrayList<>();
+			Collections.addAll(geoRel, SubjectRecipe.SubjectGeoMatchRule.GeoRelation.values());
+			if (geoRel.contains(subjectRecipe.getGeoMatchRule().geoRelation)){
+				hqlQuery += " and " + subjectRecipe.getGeoMatchRule().geoRelation.name() + "(shape, :geom) = true";
 			}else{
 				throw new IllegalArgumentException(String.format(
-						"SubjectGeoMatchRule attribute is not a valid type (is %s, can only be within)",
+						"SubjectGeoMatchRule attribute is not a valid type: ",
```
Comment (sassalley) [inline src/main/java/uk/org/tombolo/core/utils/SubjectUtils.java]: Is there any way that we can make this exception print a list of the acceptable types?

Comment (lorenaqendro) [reply src/main/java/uk/org/tombolo/core/utils/SubjectUtils.java]: Yes, there is and I will add it.

Comment (sassalley) [reply src/main/java/uk/org/tombolo/core/utils/SubjectUtils.java]: Cool - let me know once this is implemented and will approve. Would be good if @thanosbnt could look at it too.

Comment (thanosbnt) [reply src/main/java/uk/org/tombolo/core/utils/SubjectUtils.java]: I think it works great. For a future test implementation and for the sake of completeness i think we could include testing all geometries against all the others (where it makes sense).

**Comment by lorenaqendro** [comment]
@sassalley @thanosbnt thanks for the reviews. I made the suggested change. Please, approve if it looks good to merge.

**Pull request merged**